reactor-core/src/main/java/reactor/core/publisher/FluxDistinctUntilChanged.java

```diff
@@ -103,10 +111,26 @@ public boolean tryOnNext(T t) {
 				return true;
 			}
 
-			if (Objects.equals(lastKey, k)) {
+			if (null == lastKey) {
 				lastKey = k;
+				actual.onNext(t);
+				return true;
+			}
+
+			boolean equiv;
+
+			try {
+				equiv = keyComparator.test(lastKey, k);
+			}
+			catch (Throwable e) {
+				onError(Operators.onOperatorError(s, e, t));
+				return true;
+			}
+
+			if (equiv) {
```

Comment:
the current behavior is intentional. 

my use-case is emitting numerical values, and this lets me implement a "jitter" to suppress values that are "too close" to the last observed one.

if you always compared against the last key, and not the last emitted key, then in your example if the sequence kept on increasing by 1 then there would only ever be one emission.

you did find the "degenerate" case of this though, where the values dance around right below the allowable difference.